docs/standards/development_language_standards.md

```diff
@@ -71,11 +71,46 @@ application performance in most Web-facing scenarios.
 This can be seen across government as well, in particular the GDS front end community tend to publish their products
 so that they can be easily used in Ruby and Node.js, in particular.
  
-## We use JavaScript as our Node.js development language
-We do not use TypeScript (or any other extensions to the JavaScript language). 
+## We use JavaScript as our primary Node.js development language
+Teams should always look to use vanilla JavaScript in the first instance.
 
-For the types of digital services we build, TypeScript is unneccessary as any benefits don't outweigh the overhead. 
-Use vanilla JavaScript.
+A Typical Defra digital service is a simple transactional application that does not require complex type
+structures.
+
+For rarer scenarios where JavaScript alone has evidence of not being sufficient for the use case, then [JSDoc](https://jsdoc.app/) should be used to provide type annotations.
+
+Only if there is still evidence of insufficiency should [TypeScript](https://www.typescriptlang.org/) be considered.
+
+### When to use JSDoc
+- Function parameters or return values are non-trivial types
+- There is repeated ambiguity
+- Developers frequently struggle to understand how to use a function or class
+- Refactoring often causes avoidable runtime errors
+- IDE auto-completion is frequently unhelpful or misleading
+
+JSDoc should be used selectively as overuse can lead to the opposite effect, making code harder to read and maintain.
+
+### When to use TypeScript
+- Complex, deeply nested domain models that change frequently
+- Heavy use of generics or polymorphic behaviour that cannot be easily expressed in JSDoc
+- Persistent runtime errors caused by type-related issues
+- Significant refactoring risk
+- Multiple teams working on the same codebase
+- Shared libraries where consumers are in different codebases
```

Comment:
Is it worth clarifying that we would only suggest using TypeScript when there are a combination of these factors?

Just as an example, having multiple teams working on the same codebase wouldn't _necessarily_ be an indicator that TypeScript should be used.